packages/atlas-service/src/main.spec.ts

```diff
@@ -469,4 +497,98 @@ describe('AtlasServiceMain', function () {
       });
     });
   });
+
+  describe('setupAIAccess', function () {
+    beforeEach(async function () {
+      await preferencesAccess.savePreferences({
+        cloudFeatureRolloutAccess: undefined,
+      });
+    });
+
+    it('should set the cloudFeatureRolloutAccess true when returned true', async function () {
+      const fetchStub = sandbox.stub().resolves({
+        ok: true,
+        json() {
+          return Promise.resolve({
+            features: {
+              GEN_AI_COMPASS: {
+                enabled: true,
+              },
+            },
+          });
+        },
+      });
+      AtlasService['fetch'] = fetchStub;
+
+      let currentCloudFeatureRolloutAccess =
+        preferencesAccess.getPreferences().cloudFeatureRolloutAccess;
+      expect(currentCloudFeatureRolloutAccess).to.equal(undefined);
+
+      await AtlasService.setupAIAccess();
+
+      const { args } = fetchStub.getCall(0);
+
+      expect(AtlasService['fetch']).to.have.been.calledOnce;
+      expect(args[0]).to.eq(`http://example.com/ai/api/v1/hello/test`);
```

Comment:
Just FYI, you can do this shorter using `calledOnceWith`

```suggestion
      expect(AtlasService['fetch']).to.have.been.calledOnceWith('http://example.com/ai/api/v1/hello/test')
```